files/solr_check_instance.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash -eu
+# Step through all cores under one instance and throw an exit code if something is not running.
+# Exit code 5: hard fail, no need to wait any more
+# Exit code 2: possibly not loaded yet, try waiting a bit
+
+set -o pipefail
+
+if [ -z "$1" ]; then
+	echo "Cowardly refusing to reload all cores on this server. Please specify an instance as the first param."
+	exit 2
+else
+	instance="${1}v4"
+	target="*/${instance}/*"
+fi
+
+status=$(env -i curl "http://localhost:8080/$instance/admin/cores?wt=json" -sS)
+for core in $(find /var/lib/solr -maxdepth 2 -mindepth 2 -path "$target" -type d -not -name logs); do
+	if [[ -f "$core/core.properties" ]]; then
+		index=$(basename $core)
+		echo "Checking '$instance/$index'"
+		failure=$(echo "$status" | jq ".initFailures.$index")
```

Comment:
could this and the related `if [ "$failure" !=  "null" ]; then` be moved up until before the for loop? 

oh wait, it's identifying which index has an error?

Reply:
it's checking each core individually. One instance can have multiple cores, and each core can have multiple indexes.